### ISSUE TITLE
Parallel Scan hangs forever on BG/Q

### ISSUE THREAD

**Issue by stanmoore1** [kokkos/kokkos#1234]
When I run `Kokkos::parallel_scan` on more than 1 thread on BG/Q, the code hangs forever. I see this both in my own code and when running the Kokkos unit tests on BG/Q:

```
export OMP_NUM_THREADS=2
Note: Google Test filter = *scan*
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from openmp
[ RUN      ] openmp.team_scan
*Never finishes, hangs forever*


export OMP_NUM_THREADS=1
Note: Google Test filter = *scan*
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from openmp
[ RUN      ] openmp.team_scan
[       OK ] openmp.team_scan (457 ms)
[ RUN      ] openmp.scan
[       OK ] openmp.scan (1486 ms)
[ RUN      ] openmp.range_scan
[       OK ] openmp.range_scan (247 ms)
[----------] 3 tests from openmp (2192 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test case ran. (2197 ms total)
[  PASSED  ] 3 tests.
```

I'm using the latest master version of Kokkos.

**Comment by stanmoore1** [comment]
I put some `printf` statements in the `parallel_scan` in my code, and it completed a full first pass, but got hung up somewhere inside Kokkos before it came back for the final pass of the scan in my code.

**Comment by stanmoore1** [comment]
Maybe related to https://github.com/kokkos/kokkos/issues/742?

**Comment by stanmoore1** [comment]
Pretty sure this related to https://github.com/kokkos/kokkos/issues/995. The BG/Q system that I'm running on is `Big Endian`.

**Comment by stanmoore1** [comment]
PR #1237 fixes this issue.

**Comment by dsunder** [comment]
Fixed by #1240
